livekit-agents/livekit/agents/voice/agent.py

```diff
@@ -253,7 +253,9 @@ async def on_exit(self) -> None:
         """Called when the task is exited"""
         pass
 
-    async def on_end_of_turn(self, chat_ctx: llm.ChatContext, new_message: llm.ChatMessage) -> None:
+    async def on_end_of_turn(
+        self, chat_ctx: llm.ChatContext, new_message: llm.ChatMessage, generating_reply: bool
```

Comment:
I'm not a fan of having the generating_reply field here, is it needed? 
If turn_detection is set to manual, you already expect that no generation will be triggered in the current Agent

Reply:
I see, but I think it should be handled separately? This should be part of this event? 
But more like a flag saying "Continue to push audio even if allow_interruptions is False" ?